frame/support/src/traits/hooks.rs

```diff
@@ -165,27 +166,41 @@ pub trait OnRuntimeUpgrade {
 #[cfg_attr(all(feature = "tuples-96", not(feature = "tuples-128")), impl_for_tuples(96))]
 #[cfg_attr(feature = "tuples-128", impl_for_tuples(128))]
 impl OnRuntimeUpgrade for Tuple {
+	#[cfg(not(feature = "try-runtime"))]
 	fn on_runtime_upgrade() -> Weight {
 		let mut weight = Weight::zero();
 		for_tuples!( #( weight = weight.saturating_add(Tuple::on_runtime_upgrade()); )* );
 		weight
 	}
 
 	#[cfg(feature = "try-runtime")]
+	/// We are executing pre and post checks sequentially in order to be able to test seveal
+	/// consecutive migrations for the same pallet without errors. Therefore pre and post upgrade
+	/// hooks for tuples are a noop.
+	fn on_runtime_upgrade() -> Weight {
+		let mut state: Vec<u8> = Vec::default();
+		let mut weight = Weight::zero();
+		for_tuples!( #(
+			let _guard = frame_support::StorageNoopGuard::default();
+			state = Tuple::pre_upgrade().unwrap();
+			drop(_guard);
+			weight = weight.saturating_add(Tuple::on_runtime_upgrade());
+			let _guard = frame_support::StorageNoopGuard::default();
+			Tuple::post_upgrade(state).unwrap();
+			drop(_guard);
+		)* );
```

Comment:
```suggestion
		let mut weight = Weight::zero();
		for_tuples!( #(
			let _guard = frame_support::StorageNoopGuard::default();
			// expected unwrap: we want to panic if any checks fail right here right now.
			let state = Tuple::pre_upgrade().unwrap();
			drop(_guard);
			
			weight = weight.saturating_add(Tuple::on_runtime_upgrade());
			
			let _guard = frame_support::StorageNoopGuard::default();
						// expected unwrap: we want to panic if any checks fail right here right now.
			Tuple::post_upgrade(state).unwrap();
			drop(_guard);
		)* );
```

Reply:
I mean, there is no other way, the unwrap is something that has to be done as the return type is not `Result`. But I guess extra verbosity never hurt anyone.